websockets/client.py

```diff
@@ -35,9 +35,8 @@ def write_http_request(self, path, headers):
 
         """
         self.path = path
-        self.request_headers = email.message.Message()
-        for name, value in headers:
-            self.request_headers[name] = value
+        self.request_headers = http.client.HTTPMessage()
+        self.request_headers._headers = headers     # HACK
```

Comment:
Conversely, defining get/set_header at the class level is certainly more efficient that redefining them dynamically with each execution.

Historically they couldn't be methods because headers weren't stored as attributes. Certainly a good time to reconsider that part of the design.